### PR TITLE
Making it possible for the admin to view all the bundles.

### LINKED ISSUE #25127: CMS Admin should be able to view all the bundles regardless of the user who create it
### Parent Issue

_No response_

### Problem Statement

Right now Bundles are per user, so only the user who creates it can see them.

As CMS Admin Role, I should be able to view and manage all the bundles.

### Steps to Reproduce

1- Login as Chris Publisher
2- Create a bundle
3- Log out
4- Login as Admin
5- Go to the Bundles Portlet
6- No results found are shown. 

### Acceptance Criteria

Users with CMS Admin Role should be able to view/edit all the bundles.

### dotCMS Version

22.03 +

### Proposed Objective

Customer Success

### Proposed Priority

Priority 3 - Average

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

https://dotcms.zendesk.com/agent/tickets/111645

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by AndreyDotcms**
### Summary
🧪🗃️🔧

This pull request refactors the `findUnsendBundles` method of the `BundleFactoryImpl` class to allow admin users to view all the unsend bundles. It also adds a new constant `SELECT_UNSEND_BUNDLES_ADMIN` to the `BundleFactory` class and a new test class `BundleFactoryImplTest` to the integration-test package. The purpose of these changes is to improve the functionality and test coverage of the bundle business logic.

### Walkthrough
*  Add a new test class `BundleFactoryImplTest` to test the `BundleFactoryImpl` class ([link](https://github.com/dotCMS/core/pull/25949/files?diff=unified&w=0#diff-602dbe64d110f3bb7998ee89042fc7099de3b998752bd98c4b671f35950fbfe9R1-R91))
*  Add a new constant `SELECT_UNSEND_BUNDLES_ADMIN` to the `BundleFactory` class to hold a SQL query for selecting all unsend bundles ([link](https://github.com/dotCMS/core/pull/25949/files?diff=unified&w=0#diff-3c07ebcbee29abf55e99eec751e0b1b4cc84d96fa88ee535c23a53173d59d0cdR17))
*  Refactor the `findUnsendBundles` method of the `BundleFactoryImpl` class to handle the admin user scenario ([link](https://github.com/dotCMS/core/pull/25949/files?diff=unified&w=0#diff-febd349acc26740354f0be63384c3266454821c55c4066f22f3a58d59a7d2f01L12-R20), [link](https://github.com/dotCMS/core/pull/25949/files?diff=unified&w=0#diff-febd349acc26740354f0be63384c3266454821c55c4066f22f3a58d59a7d2f01L66-R92))
   *  Modify the import statements to use the `APILocator`, `DotSecurityException`, `User`, and `TestUserUtils` classes and remove the unused `PublisherConfig` class ([link](https://github.com/dotCMS/core/pull/25949/files?diff=unified&w=0#diff-febd349acc26740354f0be63384c3266454821c55c4066f22f3a58d59a7d2f01L12-R20))
   *  Add a try-catch block to load the user by the userId parameter and check if the user is a CMS admin ([link](https://github.com/dotCMS/core/pull/25949/files?diff=unified&w=0#diff-febd349acc26740354f0be63384c3266454821c55c4066f22f3a58d59a7d2f01L66-R92))
   *  Set the SQL query to either `SELECT_UNSEND_BUNDLES_ADMIN` or `SELECT_UNSEND_BUNDLES` depending on the user role and add the userId parameter if needed ([link](https://github.com/dotCMS/core/pull/25949/files?diff=unified&w=0#diff-febd349acc26740354f0be63384c3266454821c55c4066f22f3a58d59a7d2f01L66-R92))
   *  Wrap the `DotSecurityException` in a `RuntimeException` if it occurs ([link](https://github.com/dotCMS/core/pull/25949/files?diff=unified&w=0#diff-febd349acc26740354f0be63384c3266454821c55c4066f22f3a58d59a7d2f01L66-R92))

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/publisher/bundle/business/BundleFactoryImpl.java [hunk 1]
```diff
@@ -63,19 +66,30 @@ public List<Bundle> findUnsendBundles ( String userId, int limit, int offset ) t
...
+			if(APILocator.getUserAPI().isCMSAdmin(user)){
+				dc.setSQL(SELECT_UNSEND_BUNDLES_ADMIN);
+			}else {
+				dc.setSQL( SELECT_UNSEND_BUNDLES );
+				dc.addParam(userId);
+			}
+			dc.setMaxRows( limit );
+			dc.setStartRow( offset );
+
+			final List<Map<String, Object>> res = dc.loadObjectResults();
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/publisher/bundle/business/BundleFactoryImpl.java:83]: rename to resolve

File: dotCMS/src/main/java/com/dotcms/publisher/bundle/business/BundleFactoryImpl.java [hunk 2]
```diff
@@ -63,19 +66,30 @@ public List<Bundle> findUnsendBundles ( String userId, int limit, int offset ) t
...
+
+			for ( Map<String, Object> row : res ) {
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/publisher/bundle/business/BundleFactoryImpl.java:85]: set to final

**Review by jdotcms — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/publisher/bundle/business/BundleFactoryImpl.java (hunk 2 above)
Comment (AndreyDotcms) [reply dotCMS/src/main/java/com/dotcms/publisher/bundle/business/BundleFactoryImpl.java]: Done

File: dotCMS/src/main/java/com/dotcms/publisher/bundle/business/BundleFactoryImpl.java [hunk 3]
```diff
@@ -63,19 +66,30 @@ public List<Bundle> findUnsendBundles ( String userId, int limit, int offset ) t
...
-
+		try {
+			final User user = APILocator.getUserAPI().loadUserById(userId);
+			final DotConnect dc = new DotConnect();
+
+			//check if the user is a CMS Admin and depending on that, request by the owner or not
+			if(APILocator.getUserAPI().isCMSAdmin(user)){
+				dc.setSQL(SELECT_UNSEND_BUNDLES_ADMIN);
+			}else {
+				dc.setSQL( SELECT_UNSEND_BUNDLES );
```
[the remark this answers is not in the document]
Comment (AndreyDotcms) [reply dotCMS/src/main/java/com/dotcms/publisher/bundle/business/BundleFactoryImpl.java]: Done

File: dotCMS/src/main/java/com/dotcms/publisher/bundle/business/BundleFactoryImpl.java (hunk 1 above)
Comment (AndreyDotcms) [reply dotCMS/src/main/java/com/dotcms/publisher/bundle/business/BundleFactoryImpl.java]: Done

**Review by jcastro-dotcms — APPROVED** [review]

**Review by nollymar — APPROVED** [review]

**Review by erickgonzalez — APPROVED** [review]

**Pull request merged**